### PR TITLE
Fix subject name validation in TR certificate loading

### PR DESCRIPTION
**Pull request by limited**
## What does this PR (Pull Request) do?
  When validating the certificate subject name against the sslkeys hostname, TR did not properly parse Subject Alternate Names. Specifically TR assumes all subject names were contained on the Common Name (CN) line and attempted to strip off the remainder of subject before removing the wildcard.

On subject alternate names, there is no CN= delimiter and the wildcard removal was failing. 

- [x] This PR is not related to any Issue

## Which Traffic Control components are affected by this PR?
- Traffic Router
- NO Docs (just a bugfix)

## What is the best way to verify this PR?
- Run automated unit tests, specifically CertificateDataConverterTest

## The following criteria are ALL met by this PR
- [x] This PR includes tests
- [x] I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_router/connector/src/main/java/com/comcast/cdn/traffic_control/traffic_router/secure/CertificateDataConverter.java [hunk 1]
```diff
@@ -108,19 +109,28 @@ public boolean verifySubject(final X509Certificate certificate, final String hos
...
+		if (subjectName.contains("CN=")) {
+			final String[] chopped = subjectName.split("CN=", 2);
+			if (chopped != null && chopped.length > 1) {
+				final String chop = chopped[1];
+				subjectName = chop.split(",", 2)[0];
 			}
 		}
+
+		subjectName = subjectName.replaceFirst("\\*\\.", ".");
+		if (subjectName.length() > 0 && (hostAlias.contains(subjectName) || subject.contains(subjectName))) {
```
Comment (mattjackson220) [inline traffic_router/connector/src/main/java/com/comcast/cdn/traffic_control/traffic_router/secure/CertificateDataConverter.java:130]: should this maybe be `if (subjectName.length() > 0 && (hostAlias.contains(subjectName) || subjectName.contains(hostAlias))) {`? i think `subject.contains(subjectName)` will always be true

Comment (limited) [reply traffic_router/connector/src/main/java/com/comcast/cdn/traffic_control/traffic_router/secure/CertificateDataConverter.java:130]: good catch. I'll fix this and likely add a test to verify

Comment (limited) [reply traffic_router/connector/src/main/java/com/comcast/cdn/traffic_control/traffic_router/secure/CertificateDataConverter.java:130]: fixed

**Comment by rawlinp** [comment]
@limited FYI this PR has merge conflicts.

**Comment by limited** [comment]
> @limited FYI this PR has merge conflicts.

Thanks @rawlinp. 

I resolved the CHANGELOG conflict, but it might pop back up again depending on when this actually gets merged.

**Comment by mitchell852** [comment]
@limited - sorry, conflict again. :(

**Comment by limited** [comment]
fixed

**Review by mattjackson220 — APPROVED** [review]
Looks good! Tests pass and testing manually looks good!

**Pull request merged**